python/cuda_parallel/cuda/parallel/experimental/algorithms/_transform.py

```diff
@@ -0,0 +1,212 @@
+# Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.
+#
+#
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+from typing import Callable
+
+from .. import _bindings
+from .. import _cccl_interop as cccl
+from .._caching import CachableFunction, cache_with_key
+from .._utils import protocols
+from ..iterators._iterators import IteratorBase, scrub_duplicate_ltoirs
+from ..typing import DeviceArrayLike
+
+
+class _UnaryTransform:
+    __slots__ = [
+        "d_in_cccl",
+        "d_out_cccl",
+        "op_wrapper",
+        "build_result",
+    ]
+
+    def __init__(
+        self,
+        d_in: DeviceArrayLike | IteratorBase,
+        d_out: DeviceArrayLike | IteratorBase,
+        op: Callable,
+    ):
+        self.d_in_cccl = cccl.to_cccl_iter(d_in)
+        self.d_out_cccl = cccl.to_cccl_iter(d_out)
+        in_value_type = cccl.get_value_type(d_in)
+        out_value_type = cccl.get_value_type(d_out)
+        if not out_value_type.is_internal:
+            sig = (in_value_type,)
+        else:
+            sig = out_value_type(in_value_type)
+        self.op_wrapper = cccl.to_cccl_op(op, sig=sig)
+        self.build_result = cccl.call_build(
+            _bindings.DeviceUnaryTransform,
+            self.d_in_cccl,
+            self.d_out_cccl,
+            self.op_wrapper,
+        )
+
+    def __call__(
+        self,
+        d_in,
+        d_out,
+        num_items: int,
+        stream=None,
+    ):
+        cccl.set_cccl_iterator_state(self.d_in_cccl, d_in)
+        cccl.set_cccl_iterator_state(self.d_out_cccl, d_out)
```

Comment:
To avoid the overhead of Python attribute lookup, I'd suggest importing the function name into this namespace:

```
from .._cccl_interop import set_cccl_iterator_state
```

and to use it directly:

```suggestion
        set_cccl_iterator_state(self.d_in_cccl, d_in)
        set_cccl_iterator_state(self.d_out_cccl, d_out)
```

The attribute lookup was about 50-90ns in profiles I collected.